WordPress/Classes/Services/SiteTypesService.swift

```diff
@@ -0,0 +1,35 @@
+
+import Foundation
+
+
+/// Abstracts the service to obtain site types
+typealias SitesTypeServiceCompletion = (Result<[SiteType]>) -> Void
+
+protocol SiteTypesService {
+    func siteTypes(for: Locale, completion: @escaping SitesTypeServiceCompletion)
```

Comment:
I like the way you specified the broader struct here. 

We still need to figure out what our services expect, and ensure that we can provide something consistent with our Android counterparts. Perhaps it's [`languageCode`](https://developer.apple.com/documentation/foundation/locale/2293346-languagecode)?

Reply:
That indetermination is exactly why I decided to go with the Locale, as it seems broad enough to leave to the implementation of the service the decision on what to actually send up the wire